navigation/src/main/java/com/example/navigation/bottomAppNavigation/BottomAppNavigation.kt

```diff
@@ -23,47 +26,31 @@ import kotlinx.coroutines.flow.collectLatest
 @Composable
 fun BottomAppNavigation(
     ///Bottom navigation composable function that takes a
-    navigator: Navigator,
 
     ///Define the homeScreen, documentScreen, and searchScreen composable functions as parameters.
     homeScreen: @Composable () -> Unit,
     documentScreen: @Composable () -> Unit,
     searchScreen: @Composable () -> Unit
 ) {
-    val navController = rememberNavController()
 
-    ///Define a rememberDestination mutableState variable to store the current destination.
     val rememberDestination = remember { mutableStateOf(Destination.home.route) }
 
-    LaunchedEffect(Unit) {
-        navigator.actions.collectLatest {
-            when (it) {
-                is Navigator.NavigationActions.Navigate -> {
-                    navController.navigate(it.destination, builder = it.navOptions)
-                }
-
-                is Navigator.NavigationActions.Back -> {
-                    navController.popBackStack()
-                }
-            }
-        }
-    }
+    val state = rememberPagerState(initialPage = 0, pageCount = { bottomNavigationTabs.size })
```

Comment:
Using `rememberPagerState` without providing `initialPage` and `pageCount` explicitly can lead to potential issues if the `bottomNavigationTabs` size changes dynamically.

```diff
-    val state = rememberPagerState(initialPage = 0, pageCount = { bottomNavigationTabs.size })
+    val state = rememberPagerState(initialPage = 0, pageCount = bottomNavigationTabs.size)
```